### PR TITLE
Faisal

### PR DESCRIPTION
**Pull request by mfaisalnazari**
iteration 3 complete

### CODE REVIEW

File: lib/facility.rb [hunk 1]
```diff
@@ -1,14 +1,74 @@
...
+        vehicle.plate_type = :antique
+      elsif vehicle.electric_vehicle?
+          @collected_fees += 200
+          vehicle.plate_type = :ev
+      else 
+        @collected_fees += 100
+        vehicle.plate_type = :regular
+        end
+      @registered_vehicles << vehicle
+    end
```
Comment (jamisonordway) [inline lib/facility.rb:33]: For nested conditionals like this one, you could use an explicit return for a guard clause that would eliminate the need for nesting:
```ruby
return false unless @services.include?('Vehicle Registration')
vehicle.registration_date = Time.now
# etc
```

File: lib/facility_factory.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class FacilityFactory
+    
+    def create_facilities(facility_details)
+        
+        facilities_created = []
+        facility_details.each do |facility|
+          facility1 = {}
```
Comment (jamisonordway) [inline lib/facility_factory.rb:7]: I might call this something like `facility_data`, `facility1` feels arbitrary and less dynamic than it actually is.

File: lib/vehicle.rb [hunk 3]
```diff
@@ -6,13 +6,19 @@ class Vehicle
...
+              
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @registeration_date 
+    @plate_type
```
Comment (jamisonordway) [inline lib/vehicle.rb:21]: It seems like this is working as is, but I would expect these attributes to be defined with an assignment operator, even if the values start as nil or an empty string.
```ruby
@registration_date = nil
@plate_type = nil
```

File: lib/vehicle_factory.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class VehicleFactory
+    
+    def create_vehicles(vehicle_details)
+        vehicles_created = []
+        vehicle_details.each do |vehicle|
+            vehicle[:engine] = "ev"
+            vehicles_created << Vehicle.new(vehicle)
+        end
+        vehicles_created
```
Comment (jamisonordway) [inline lib/vehicle_factory.rb:9]: You could refactor this to use `.map` and shorten your code:
```ruby
def create_vehicles(vehicle_details)
  vehicle_details.map do |vehicle|
    vehicle[:engine] = "ev"
    Vehicle.new(vehicle)
  end
end
```

File: spec/facility_spec.rb [hunk 5]
```diff
@@ -22,5 +27,64 @@
...
+      expect(@cruz.registration_date).to be_within(1.0).of(Time.now)
+      expect(@cruz.plate_type).to eq(:regular)
+      expect(@facility.registered_vehicles).to eq([@cruz])
+      expect(@facility.collected_fees).to eq(100)
+
+    end
+    
+  end
+  describe 'written test' do
+    it 'can take test' do
```
Comment (jamisonordway) [inline spec/facility_spec.rb:44]: Generally it's a better practice to use `true` and `false` instead of `!false` and `!true`. These tests would work the same way with this change.
If you really need to test for any false-y value (in Ruby, the only false-y values are `nil` and `false`), you could do something like `expect(<whatever>).to be_falsey`, or for the opposite, `expect(<whatever>).to be_truthy`

File: spec/registrant_spec.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    end
+    describe 'attributes' do
+        it 'has attributes' do
+            expect(@registrant_1.name).to eq("Bruce")
+            expect(@registrant_1.age).to eq(18)
+            expect(@registrant_1.permit).to eq(true)
+            expect(@registrant_1.license_data).to eq({:license=>false, :renewed=>false, :written=>false})
+        end
+    end
+    describe 'methods work' do
```
Comment (jamisonordway) [inline spec/registrant_spec.rb:21]: I would call this describe `#earn_permit` or maybe `#earn_permit changes return value of #permit?`
